lib/check/single_module_file.ex

```diff
@@ -0,0 +1,66 @@
+defmodule CompassCredoPlugin.Check.SingleModuleFile do
+  use Credo.Check,
+    base_priority: :high,
+    category: :refactor,
+    explanations: [
+      check: """
+        Prefer one module per source file.
+
+        # lib/account/user.ex
+        defmodule App.Accounts.User do
+          defmodule App.Accounts.SuperUser do
+            defstruct name: "John", age: 27
+          end
+
+          # ...
+        end
+
+        # lib/account/user.ex
+        defmodule App.Accounts.User do
+          # ...
+        end
+
+        # lib/account/super_user.ex
+        defmodule App.Accounts.SuperUser do
+          defstruct name: "John", age: 27
+
+          # ...
+        end
+      """
+    ]
+
+  def run(%SourceFile{} = source_file, params) do
+    issue_meta = IssueMeta.for(source_file, params)
+
+    issues =
+      source_file
+      |> Credo.Code.to_tokens()
+      |> Enum.filter(fn line ->
+        case line do
+          {:identifier, _meta, :defmodule} -> true
+          _ -> false
+        end
+      end)
+      |> Enum.reduce([], fn item, acc ->
+        case item do
+          {:identifier, {line, _columns, _}, _} ->
+            [issue_for(issue_meta, line, "") | acc]
+        end
+      end)
```

Comment:
I think we can merge filter and reduce them into a single reduce.
```suggestion
      |> Enum.reduce([], fn item, acc ->
        case item do
          {:identifier, {line, _columns, _}, : defmodule} ->
            [issue_for(issue_meta, line, "") | acc]
            
            _ -> acc
        end
      end)
```

Reply:
thanks, updated in https://github.com/nimblehq/compass-credo-plugin/pull/16/commits/1200880a7096bb7af3c614e5844cbe003b640dd1